src/pages/Editor/Classes/UpdateClassModals/EditClassModal.tsx

```diff
@@ -0,0 +1,165 @@
+/* eslint-disable no-bitwise */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+
+import React, { useEffect, useState } from 'react';
+import { useFieldArray, useForm } from 'react-hook-form';
+import styled from 'styled-components';
+import { Input, InputField, TextArea } from 'components/Input';
+import { ErrorMessageInterface, EditClassModalProps } from 'types';
+import { CreateClassProps, editClass } from 'api/classes';
+
+import { FormErrorMessages, StyledErrorMessage } from 'components/ErrorMessage';
+
+import { Modal } from 'components/Modal/Modal';
+import { Button, PrimaryButton } from 'components/Button';
+import { classValidationRules } from 'utilities/validation';
+import { FaPlus } from 'react-icons/fa';
+import { TrashIcon } from 'pages/Editor/Classes/ClassComponent.styled';
+
+const StyledForm = styled.form`
+  display: flex;
+  flex-direction: column;
+  gap: 1em;
+`;
+
+const Classworks = styled.div`
+  background: #fff;
+  border-radius: 4px;
+  padding: 1em;
+  overflow: hidden;
+  box-shadow: 0 13px 27px -5px hsla(240, 30.1%, 28%, 0.25),
+    0 8px 16px -8px hsla(0, 0%, 0%, 0.3), 0 -6px 16px -6px hsla(0, 0%, 0%, 0.03);
+`;
+
+/* eslint no-underscore-dangle: 0 */
+
+export const EditClassModal = ({
+  item,
+  isOpen,
+  toggle,
+  fetchClasses,
+}: EditClassModalProps) => {
+  const {
+    register,
+    control,
+    setValue,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<CreateClassProps>({
+    defaultValues: {
+      classworks: item.classworks,
+    },
+    mode: 'onChange',
+  });
+  const { fields, append, remove } = useFieldArray({
+    name: 'classworks',
+    control,
+  });
+  useEffect(() => {
+    setValue('name', item.name);
+    setValue('description', item.description);
+  }, [item.name, item.description, setValue]);
+
+  const [errorMessages, setErrorMessages] = useState<ErrorMessageInterface[]>(
+    []
+  );
+
+  // ClassWork
+
+  const handleCancelModal = () => {
+    setValue('name', item.name);
+    setValue('description', item.description);
+    reset();
```

Comment:
what does this reset do?
I think these two function calls can be combined together as reset
```typescript
setValue('name', item.name);
setValue('description', item.description);
```

Reply:
reset is an inbuilt method available from React-hook-form. I refactored it though :)